src/index.js

```diff
@@ -32,43 +33,39 @@ const {
   staleWhileRevalidate,
   staleIfError
 } = config.get('caching');
+const { homePage } = config.get('service');
 
 const cacheHeader =
   cachingEnabled &&
   `public, must-revalidate, max-age=${browserMaxAge}, s-maxage=${cdnMaxAge}, stale-while-revalidate=${staleWhileRevalidate}, stale-if-error=${staleIfError}`;
 
-const app = express();
-app.set('etag', true);
-app.use(compression());
-
-function handleBrowsersBadge(req, res, browsersList) {
+// ============================================================================
+// Helpers
+// ============================================================================
+// Complete response body for a browsers badge.
+const getBrowsersBadgeBody = async ({ req, getBrowsers }) => {
   const query = { style: req.query.style };
-  // eslint-disable-next-line promise/catch-or-return
-  Promise.resolve(browsersList)
-    .then(browsers => {
-      if (browsers.length) {
-        const options = {
-          logos: req.query.logos,
-          labels: req.query.labels,
-          exclude: req.query.exclude,
-          sortBy: req.query.sortBy,
-          versionDivider: req.query.versionDivider,
-          ...query
-        };
-        return getBrowsersBadge({ browsers, options });
-      }
-      return getShieldsBadge('browsers', 'unknown', 'lightgrey', query);
-    })
-    .catch(err => {
-      console.error(`Error: ${err}`);
-      return getShieldsBadge('browsers', 'unknown', 'lightgrey', query);
-    })
-    // eslint-disable-next-line promise/always-return
-    .then(body => {
-      res.write(body);
-      res.end();
-    });
-}
+  const options = {
+    logos: req.query.logos,
+    labels: req.query.labels,
+    exclude: req.query.exclude,
+    sortBy: req.query.sortBy,
+    versionDivider: req.query.versionDivider,
+    ...query
+  };
+
+  let browsers;
+  try {
+    browsers = await Promise.resolve(getBrowsers);
```

Comment:
Do we need the `Promise.resolve` here? I think `getBrowsers` is already a Promise, so we can just `await` it?

Reply:
If my reading was correct, it can be an object as well. I found this in index.js:

```js
const getBrowsers = getGroupedBrowsers(browsers);
```

and `getGroupedBrowsers` returns an object.

Reply:
Aha, that makes sense now!